### PR TITLE
Updated customer extension module template with lasted notes

### PR DESCRIPTION
**Pull request by evanbtlr**

### CODE REVIEW

File: CSharp/VirtoCommerce.MemberExModule.Template/Core/Models/SupplierReview.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
-﻿using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Common;
 
 namespace $safeprojectname$.Models
 {
-    public class SupplierReview : AuditableEntity
-    {
+    public class SupplierReview : Entity
+{
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Core/Models/SupplierReview.cs]: bad formatting. You should configure "Power Commands for Visual Studio"

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs [hunk 2]
```diff
@@ -39,6 +36,7 @@ public override void Down()
             DropForeignKey("dbo.MemberSupplier", "Id", "dbo.Member");
             DropForeignKey("dbo.MemberSupplierReview", "SupplierId", "dbo.MemberSupplier");
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs]: remove empty lines

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-detail.js [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
 
-            blade.fillDynamicProperties();
-        } else {
-            blade.subtitle = 'Supplier details';
-        }
+            if (blade.isNew) {
+                blade.title = 'external-customer.blades.supplier-detail.title';
+                blade.currentEntity = angular.extend({
+                    reviews: []
+                }, blade.currentEntity);
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-detail.js]: remove 3 redundant lines. Check, if that doesn't break everything

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/widgets/supplier-reviews-widget.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿<div class="gridster-cnt" ng-click="openBlade()">
+    <div class="cnt-inner">
+        <div class="list-count">{{blade.currentEntity.reviews.length | number:0}}</div>
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/widgets/supplier-reviews-widget.html]: should be fine without " | number:0"

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-detail.js [hunk 5]
```diff
@@ -1,23 +1,24 @@
 angular.module('$ext_safeprojectnamecamel$')
-    .controller('$ext_safeprojectnamecamel$.supplierDetailController', ['$scope', function ($scope) {
-var blade = $scope.blade;
-        
-        if (blade.isNew) {
-            blade.title = 'New Supplier';
-            blade.currentEntity = angular.extend({
-                reviews: []
-            }, blade.currentEntity);
+    .controller('$ext_safeprojectnamecamel$.supplierDetailController', ['$scope', 'virtoCommerce.customerModule.members',
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-detail.js]: remove redundant reference to 'virtoCommerce.customerModule.members'

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-review-list.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
+angular.module('$ext_safeprojectnamecamel$')
+    .controller('$ext_safeprojectnamecamel$.reviewsController', ['$scope', 'platformWebApp.bladeNavigationService',
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/blades/supplier-review-list.js]: rename to ...supplierReviewListController

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/widgets/supplier-reviews-widget.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+angular.module('$ext_safeprojectnamecamel$')
+    .controller('$ext_safeprojectnamecamel$.reviewsWidgetController',
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/widgets/supplier-reviews-widget.js]: rename to match file name

File: CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/module.js [hunk 8]
```diff
@@ -5,17 +5,47 @@ if (AppDependencies !== undefined) {
     AppDependencies.push(moduleName);
 }
 
-angular.module(moduleName, [])
-    .run(['$rootScope', 'virtoCommerce.customerModule.memberTypesResolverService',
-        function ($rootScope, memberTypesResolverService) {
+angular.module(moduleName, ['virtoCommerce.customerModule'])
+    .run(['$rootScope', 'platformWebApp.widgetService', 'virtoCommerce.customerModule.memberTypesResolverService',
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Web/Scripts/module.js]: remove redundant reference to rootScope

**Review by eMazeika — CHANGES_REQUESTED** [review]
Change 
// Here you can write code for custom mapping
// supplier properties will be mapped in base method implementation by using value injection
to
// Here you can write code for custom mapping.
// Member properties will be mapped in base method implementation by using value injection
Also, move the comment to correct location: before custom mapping code begins.

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs [hunk 9]
```diff
@@ -42,17 +42,17 @@ public override MemberDataEntity FromModel(Member member, PrimaryKeyResolvingMap
                 }
             }
 
-            // Here you can write code for custom mapping
-            // supplier properties will be mapped in base method implementation by using value injection
+            // Here you can write code for custom mapping.
+            // Member properties will be mapped in base method implementation by using value injection
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs]: move comment before line 31

**Review by eMazeika — CHANGES_REQUESTED** [review]

**Review by eMazeika — CHANGES_REQUESTED** [review]
rename var suplierDataEntity to
 var supplierDataEntity

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs (hunk 2 above)
Comment (evanbtlr) [reply CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs]: Extra empty line removed. The rest too to delete? (Other templates also use spaces in such situations)

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs (hunk 9 above)
Comment (evanbtlr) [reply CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs]: moved

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs (hunk 2 above)
Comment (eMazeika) [reply CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs]: also remove empty line 32. And yes, remove empty lines in all templates

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs [hunk 10]
```diff
@@ -42,39 +44,39 @@ public override MemberDataEntity FromModel(Member member, PrimaryKeyResolvingMap
...
         public override Member ToModel(Member member)
         {
-            // Here you can write code for custom mapping
-            // supplier properties will be mapped in base method implementation by using value injection
+            // Here you can write code for custom mapping.
+            // Member properties will be mapped in base method implementation by using value injection
             var retVal = base.ToModel(member) as Supplier;
 
             if (retVal != null)
                 retVal.Reviews = Reviews.OrderBy(x => x.Id).Select(x => x.ToModel(AbstractTypeFactory<SupplierReview>.TryCreateInstance())).ToList();
```
Comment (eMazeika) [inline CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs]: why OrderBy(..) is used? Any reason to sort?

Comment (evanbtlr) [reply CSharp/VirtoCommerce.MemberExModule.Template/Data/Models/SupplierDataEntity.cs]: Yes you are right. Fixed.

File: CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs (hunk 2 above)
Comment (evanbtlr) [reply CSharp/VirtoCommerce.MemberExModule.Template/Data/Migrations/201906110802132_Initial.cs]: Good, deleted

**Review by eMazeika — CHANGES_REQUESTED** [review]
set version 2.2.0

**Pull request merged**